lib/CL/pocl_ipc_mutex.h

```diff
@@ -0,0 +1,77 @@
+/* pocl_ipc_mutex.h: Named interprocess mutex
+
+   Copyright (c) 2024 Henry Linjam�ki / Intel Finland Oy
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+/* This header hosts API for named interprocess mutex that does not
+ * cause deadlock if a process dies or exits without unlocking
+ * first. Beware that deadlocks are still possible within process if a
+ * thread forgets to unlock the mutex.  */
+
+#ifndef POCL_IPC_MUTEX_H
+#define POCL_IPC_MUTEX_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+typedef struct pocl_ipc_mutex_s
+{
+  void *handle;
+} pocl_ipc_mutex_t;
+
+/** Creates a mutex by 'name' or opens an existing one.
+ *
+ * On success, the return value is zero and a valid mutex object returned
+ * via 'ipc_mtx'.  */
+int pocl_ipc_mutex_create (const char *name, pocl_ipc_mutex_t *ipc_mtx);
+
+/** Lock the mutex.
+ *
+ * The call is blocked if another process or thread by the same name
+ * has locked the mutex. The mutex is unlocked in case the process
+ * holding the lock terminates without unlocking first.
+ *
+ * The '�pc_mtx' must be a valid object created by pocl_ipc_mutex_create().
+ *
+ * If mutex is locked successfully, zero is returned. Otherwise, on an error,
+ * non-zero is returned.  */
+int pocl_ipc_mutex_lock (pocl_ipc_mutex_t ipc_mtx);
+
+/** Combined variant of pocl_ipc_mutex_create() and pocl_ipc_mutex_lock().  */
+int pocl_ipc_mutex_create_and_lock (const char *name,
+                                      pocl_ipc_mutex_t *ipc_mtx);
```

Comment:
Check the style with formatter scripts. I think the C style wants the return values in their own line, etc.

Reply:
Fixed.

> I think the C style wants the return values in their own line, etc.

I ran `git clang-format-17 --binary /usr/bin/clang-format-17  --style=file:./tools/scripts/style.GNU main -- *.c *.h` but it didn't want to do this.

Reply:
Also the second line seems badly indented. Can you run the clang-format-branch.sh or similar script?

Reply:
Fixed manually. 

> Can you run the clang-format-branch.sh or similar script?

No improvement over the `git clang-format ...`.